src/core/qgsmaplayer.cpp

```diff
@@ -1980,3 +1990,41 @@ void QgsMapLayer::onNotifiedTriggerRepaint( const QString &message )
   if ( refreshOnNotifyMessage().isEmpty() || refreshOnNotifyMessage() == message )
     triggerRepaint();
 }
+
+QgsRectangle QgsMapLayer::wgs84Extent( bool forceRecalculate ) const
+{
+  QgsRectangle wgs84Extent;
+
+  if ( ! forceRecalculate && ! mWgs84Extent.isNull() )
+  {
+    wgs84Extent = mWgs84Extent;
+  }
+  else
+  {
+    const QgsCoordinateTransform transformer { crs(), QgsCoordinateReferenceSystem::fromOgcWmsCrs( geoEpsgCrsAuthId() ), transformContext() };
```

Comment:
QgsProject::setTransformContext will need to invalidate mWgs84Extent

Reply:
I didn't want to add a specific method in the public API so each subclasses of `QgsMapLayer` needs to call a protected `invalidateWgs84Extent()` in `setTransformContext()`.

Let me know if you think otherwise.

Reply:
Sounds good to me.